### PR TITLE
Support transmission in translation from Standard Surface to glTF PBR

### PR DESCRIPTION
**Pull request by charo-m**
This corrects the translation of transmission parameters from Standard Surface to glTF PBR.
Issue: https://github.com/AcademySoftwareFoundation/MaterialX/issues/1599

Here's the change in a more readable pseudocode:
```
if (transmission > 0) {
    if (transmission_depth > 0) {
        attenuation_color = transmission_color
        base_color = white
    } else {
       attenuation color = white
       base_color = transmission_color
    }
} else {
    attenuation_color = white (default)
    base_color = no_transmission_base_color (hasCoatColor ? mixedBaseColor : scaledBaseColor)
}

```
Before ->
BeforeGraph

After ->
AfterGraph

### CODE REVIEW

**Comment by jstone-lucasfilm** [comment]
@charo-m This looks really promising, and to resolve the EasyCLA issue, I'd recommend following up through the support link that the Linux Foundation provides:

https://jira.linuxfoundation.org/servicedesk/customer/portal/4

**Comment by charo-m** [comment]
thanks @jstone-lucasfilm ! I had a mismatching email address, just managed to get my CLA signed.

**Comment by jstone-lucasfilm** [comment]
@charo-m Excellent!  Sometimes the LFX needs to bump the EasyCLA check so that it updates to your new email address, so feel free to check with them if it doesn't automatically turn from red to green.

**Comment by kwokcb** [comment]
As a suggestion, it might might be better test to have the graph with the translation node in it to be able to do visual compares as part of the render test suite. Perhaps it be added to the test suite in a new folder `resources/Materials/TestSuite/pbrlib/translation`. ( @jstone-lucasfilm, I don't see any other translation examples anywhere else ?  Others could be added in as desired).

File: libraries/bxdf/translation/standard_surface_to_gltf_pbr.mtlx [hunk 1]
```diff
@@ -76,12 +76,18 @@
     </dotproduct>
 
     <!-- Metallic roughness -->
-    <ifequal name="base_color" type="color3">
+    <ifequal name="no_transmission_base_color" type="color3">
```
Comment (jstone-lucasfilm) [inline libraries/bxdf/translation/standard_surface_to_gltf_pbr.mtlx:79]: For consistency with other example graphs, I would recommend using camel case names for nodes representing intermediate variables, with snake case used only for the final nodes that match the naming conventions in glTF PBR.

So this variable would be named `noTransmissionBaseColor`, and similarly the new intermediate variables below would be named `transmissionInEffectAttenuationColor` and so forth.

File: source/MaterialXTest/MaterialXGenShader/GenShader.cpp [hunk 2]
```diff
@@ -141,6 +141,43 @@ TEST_CASE("GenShader: Shader Translation", "[translate]")
     }
 }
 
+TEST_CASE("GenShader: Gltf Transmission Translation", "[translate]")
```
Comment (jstone-lucasfilm) [inline source/MaterialXTest/MaterialXGenShader/GenShader.cpp:144]: For now, I would recommend omitting this new unit test, as what we really want is _render validation_ for a wider set of shader translation use cases, including the new case for which you've added support.

At the moment, we have just one shader translation test in our GitHub Actions suite, as we're currently using CPU renders and the overhead of each new test is quite high, but this is the location where we'll want to add new tests when GPU renders are supported:

https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/.github/workflows/main.yml#L265

**Review by jstone-lucasfilm — COMMENTED** [review]
This looks really good, thanks @charo-m, and I had just two recommendations for improvements.

**Comment by jstone-lucasfilm** [comment]
Thanks for these updates, @charo-m!

For now, I think it's fine to omit the new `standard_surface_to_gltf_pbr_glass.mtlx` unit test, which is an interesting suggestion for a future improvement to our translation system, but not an update that we need as part of your Dev Days project.

**Review by jstone-lucasfilm — APPROVED** [review]
Great work on this improvement, thanks @charo-m!

**Pull request merged**